use-gui/src/main/java/org/tzi/use/gui/main/EvalOCLDialog.java

```diff
@@ -315,4 +359,117 @@ private boolean evaluate(String in, boolean evalTree) {
+     *
+     * @param suggestion The {@code SuggestionResult} object containing the list of suggestions and the prefix.
+     * @param textPane    The {@code JPanel} where the autocomplete suggestions will be displayed.
+     */
+    private void displayResultsWindow(SuggestionResult suggestion, JPanel textPane) {
+        List<String> suggestionList = suggestion.suggestions;
+        List<String> displayedList = formatStrings(suggestion, suggestionList);
+
+        if (autocompletionScrollPane == null) {
+            autocompletionResultList = new JList<>(displayedList.toArray(new String[0]));
+            autocompletionResultList.setSelectionMode(ListSelectionModel.SINGLE_SELECTION);
+            autocompletionResultList.setCellRenderer(new TwoColorListCellRenderer());
+
+            autocompletionResultList.getInputMap(JComponent.WHEN_FOCUSED).put(KeyStroke.getKeyStroke(KeyEvent.VK_ENTER, 0), "confirmSelection");
+            autocompletionResultList.getActionMap().put("confirmSelection", new AbstractAction() {
+                @Override
+                public void actionPerformed(ActionEvent e) {
+                    String selectedValue = suggestionList.get(autocompletionResultList.getSelectedIndex());
+                    if (selectedValue != null) {
+                        String currentText = fTextIn.getText();
+
+                        String newText;
+                        boolean hasParenthesis = false;
+                        if (selectedValue.endsWith(")")) {
+                            if (selectedValue.contains("(")) {
+                                int firstIndex = selectedValue.indexOf("(");
+                                selectedValue = selectedValue.substring(0, firstIndex + 1);
+                            }
+                            newText = currentText + selectedValue + ")";
+                            hasParenthesis = true;
+                        } else if(currentText.endsWith(")")){
+                            newText = currentText.substring(0, currentText.length()-1) + selectedValue + ")";
+                            hasParenthesis = true;
+                        } else {
+                            newText = currentText + selectedValue;
+                        }
+
+                        fTextIn.setText(newText);
+                        if(hasParenthesis){
+                            fTextIn.setCaretPosition(newText.length()-1);
+                        } else {
+                            fTextIn.setCaretPosition(newText.length());
+                        }
+                    }
+                    autocompletionScrollPane = null;
+                    autocompletionResultList = null;
+                    textPane.remove(1);
+                    fTextIn.requestFocus();
+                    pack();
+                }
+            });
+
+            autocompletionScrollPane = new JScrollPane(autocompletionResultList);
+            textPane.add(autocompletionScrollPane, 1);
```

Comment:
Yes good idea. Since 06c5b06713b64fa6c4f0cb5259c74164d636a6ca the auto completion windows depend on the caret position and the prefix already present in the input text area. In my opinion, this resolves the issue effectively.